weston-ivi-shell/src/ivi-share.h

```diff
@@ -44,6 +51,7 @@ struct ivi_share_nativesurface
     uint32_t send_flag;
     struct wl_listener surface_destroy_listener;
     struct ivi_shell_share_ext *shell_ext;
+    struct ivi_share_buffer_reference buffer_refs[2], *current, *back;
```

Comment:
why do you need a static array and two pointers, which points contents of the array all the time? 
you can just  have two buffer members:
struct ivi_share_buffer_reference current, back;

Reply:
"current" and "back" need to swap. Please find ivi-share.c L305 for the detail.
It is needed to avoid direct value copy of structure. If the copy cost can be accepted, it can be modified to your suggestion.